app/controllers/UnloadingPermissionController.scala

```diff
@@ -48,147 +35,24 @@ class UnloadingPermissionController @Inject()(
 )(implicit ec: ExecutionContext)
     extends BackendController(cc) {
 
-  private val logger = Logger(getClass)
-
-  private def logErrors(errors: NonEmptyChain[ValidationError]): Unit =
-    logger.warn(errors.toChain.toList.mkString("\n"))
-
-  def post(): Action[PermissionToStartUnloading] = Action.async(parse.json[PermissionToStartUnloading]) {
+  def get(): Action[NodeSeq] = Action.async(parse.xml) {
     implicit request =>
-      conversionService.convertUnloadingPermission(request.body).map {
-        _.fold(
-          errors => {
-            logErrors(errors)
-
-            errors match {
-              case x if x.exists(y => y.isInstanceOf[JsonError])                   => InternalServerError
-              case x if x.exists(y => y.isInstanceOf[ReferenceDataRetrievalError]) => BadGateway
-              case _                                                               => BadRequest
+      XmlReader
+        .of[PermissionToStartUnloading]
+        .read(request.body)
+        .toOption
+        .map {
+          permission =>
+            conversionService.convertUnloadingPermission(permission).map {
```

Comment:
This is starting to look like something that we might want to move out of the controller layer, which would simplify the test setup